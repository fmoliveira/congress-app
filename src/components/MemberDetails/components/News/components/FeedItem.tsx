import React from "react"
import styled from "styled-components"

import { Badge, Date } from "../../../../_common"
import { IFeedItem } from "../types"

const ItemWrapper = styled.div`
  display: grid;
  grid-template-columns: 100px 1fr;
  align-items: center;
  margin: 0.5em 0;
  color: #888888;
`

const Link = styled.a`
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  font-size: 0.9em;
  color: #999999;
  text-decoration: none;

  &:hover {
    text-decoration: underline;
  }
`

interface IProps {
  feed: IFeedItem
}

export const FeedItem = ({ feed }: IProps) => (
  <ItemWrapper>
    <Badge>
      <Date value={feed.pubDate} displayFormat="MMM D, YYYY" />
    </Badge>
    <Link href={feed.link} target="_blank" rel="noopener noreferrer">
      {feed.title}
    </Link>
  </ItemWrapper>
)
