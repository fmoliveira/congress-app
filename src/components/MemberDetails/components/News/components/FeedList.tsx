import React from "react"

import { size } from "lodash"
import shortId from "shortid"

import { IFeedItem } from "../types"
import { FeedItem } from "./index"

interface IProps {
  feed: IFeedItem[]
}

const FeedList = ({ feed = [] }: IProps) => (
  <div>
    {size(feed) === 0 && <div>No news shared recently.</div>}
    {feed.map(i => (
      <FeedItem key={shortId.generate()} feed={i} />
    ))}
  </div>
)

export default FeedList
