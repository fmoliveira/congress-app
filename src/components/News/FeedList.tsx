import React from "react"

interface IProps {
  feed: any[]
}

export const FeedList = ({ feed = [] }: IProps) => (
  <div>{feed.length} items</div>
)
