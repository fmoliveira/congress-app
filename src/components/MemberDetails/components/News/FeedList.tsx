import React from "react"

import { FeedItem } from "./FeedItem"

interface IProps {
  feed: any[]
}

export const FeedList = ({ feed = [] }: IProps) => (
  <div>
    {feed.map(i => (
      <FeedItem key={i.key} feed={i} />
    ))}
  </div>
)
