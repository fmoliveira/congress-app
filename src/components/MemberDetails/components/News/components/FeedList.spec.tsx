import React from "react"
import { cleanup, render } from "react-testing-library"

import FeedItem from "./FeedItem"
import FeedList from "./FeedList"

jest.mock("./FeedItem", () => jest.fn(() => <div>FeedItem</div>))

const setup = (props?: any) => render(<FeedList {...props} />)

describe("FeedList", () => {
  afterEach(() => {
    cleanup()
    jest.clearAllMocks()
  })

  it("Should render without crashing", () => {
    const { container } = setup()
    expect(container.firstChild).toBeTruthy()
    expect(container).toMatchSnapshot()
  })

  it("Should not render FeedItem instances for empty feed", () => {
    setup()
    expect(FeedItem).not.toHaveBeenCalled()
  })

  it("Should render a FeedItem instance", () => {
    const props = { feed: [{ foo: "bar" }] }
    setup(props)

    expect(FeedItem).toHaveBeenCalledTimes(1)
    expect(FeedItem).toHaveBeenCalledWith({ feed: { foo: "bar" } }, {})
  })

  it("Should render multilple FeedItem instances", () => {
    const props = {
      feed: [{ foo: "bar" }, { baz: "inga" }, { trust: "more" }]
    }
    setup(props)

    expect(FeedItem).toHaveBeenCalledTimes(3)
    expect(FeedItem).toHaveBeenNthCalledWith(1, { feed: { foo: "bar" } }, {})
    expect(FeedItem).toHaveBeenNthCalledWith(2, { feed: { baz: "inga" } }, {})
    expect(FeedItem).toHaveBeenNthCalledWith(3, { feed: { trust: "more" } }, {})
  })
})
