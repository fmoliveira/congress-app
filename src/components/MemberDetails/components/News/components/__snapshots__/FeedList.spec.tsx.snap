// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedList Should render without crashing 1`] = `
<div>
  <div>
    <div>
      No news shared recently.
    </div>
  </div>
</div>
`;
